doc/guide/cockpit-dbus.xml

```diff
@@ -853,25 +853,55 @@ client.onmeta = function(ev, data) { ... }
     <para>The <code>data</code> has the following form:</para>
 
 <programlisting>
-{
+  {
       "org.Interface": {
           "methods": {
-              "Method1": { },
+              "Method1": {
+                  "in": [ "s", "v" ],
+                  "out": [ "i" ]
+              },
               "Method2": { }
           },
+          "signals": {
+              "Signal": {
+                  "in": [ "b", "s" ]
+              }
+          },
           "properties": {
-              "Prop1": { "flags": "rw" },
-              "Prop2": { "flags": "r" }
+              "Prop1": {
+                  "flags": "rw",
+                  "type": "s"
+              },
+              "Prop2": {
+                  "flags": "r",
+                  "type": "b"
+              }
           }
       }
-}
+  }
 </programlisting>
 
     <para>Multiple interfaces may be present, each of which may have methods and properties.
       This is emitted before the first <link linkend="cockpit-dbus-proxy"><code>client.onnotify</code></link>
       event for the relevant interface.</para>
   </refsection>
 
+  <refsection id="cockpit-dbus-meta">
+    <title>client.meta()</title>
+<programlisting>
+client.meta(data)
+</programlisting>
+    <para>Populate DBus introspection metadata about DBus interfaces. Usually this metadata
+      is automatically introspected from the DBus services called by the <code>client</code>,
+      but in certain cases it is useful to populate this info from javascript code. One such
+      case is when exporting a DBus service from javascript.</para>
```

Comment:
I don't think so. Some services are broken, and this is another way to backfill data for such services.